### PR TITLE
Fix appstream XML and use appstreamcli for validation

### PR DESCRIPTION
**Pull request by phw**
# Summary

* This is a…
  * [x] Bug fix
  * [ ] Feature addition
  * [ ] Refactoring
  * [ ] Minor / simple change (like a typo)
  * [ ] Other
* **Describe this change in 1-2 sentences**:

# Problem

The AppStream metadata was invalid (multiple `<recommends>` , should be only one).

We have AppStream validation already. But it was using the older ` appstream-util` which did not catch that issue. Replaced this with `appstreamcli`.

### CODE REVIEW

File: .github/workflows/validate-appstream.yml [hunk 1]
```diff
@@ -19,11 +19,11 @@ jobs:
...
-    - name: Install appstream-util
+    - name: Install appstream
       run: |
         sudo apt-get update
         sudo apt-get install appstream-util gettext
     - name: Validate AppStream metadata
       run: |
         python setup.py build_appdata
-        appstream-util validate-relax org.musicbrainz.Picard.appdata.xml
+        appstreamcli validate --pedantic --explain org.musicbrainz.Picard.appdata.xml
```
Comment (zas) [inline .github/workflows/validate-appstream.yml:29]: https://manpages.ubuntu.com/manpages/jammy/en/man1/appstreamcli.1.html says `appstreamcli` is provided by `appstream` package, which is not installed above.

**Review by zas — CHANGES_REQUESTED** [review]

File: .github/workflows/validate-appstream.yml [hunk 2]
```diff
@@ -19,11 +19,11 @@ jobs:
       uses: actions/setup-python@v4
       with:
         python-version: '3.11'
-    - name: Install appstream-util
+    - name: Install appstream
       run: |
         sudo apt-get update
         sudo apt-get install appstream-util gettext
```
Comment (zas) [inline .github/workflows/validate-appstream.yml:25]:
```suggestion
        sudo apt-get install appstream gettext
```

Not sure appstream-util is still needed

**Review by zas — CHANGES_REQUESTED** [review]

File: .github/workflows/validate-appstream.yml (hunk 2 above)
Comment (phw) [reply .github/workflows/validate-appstream.yml:25]: I wanted to change that, but only changed it in the step's description. Oh well, will fix.

**Comment by phw** [comment]
@zas Validation fails, but that's because of appstream 0.15 on the CI. Locally with 0.16 it validates (with some "pedantic" output I keep for information, but it doesn't cause the validation to fail).

I'll revert this back to appstream-cli for now.

**Comment by phw** [comment]
@zas https://github.com/metabrainz/picard/pull/2241/commits/bbd98b97d4f75a3566f41bf2d34b59663a2aed5d

**Review by zas — APPROVED** [review]

**Pull request merged**